### DISCUSSION TITLE
"Error constructing OIDC authentication configuration - FetchError" when using OIDC authentication

### DISCUSSION THREAD

**Discussion by dpfr2022** [Budibase/budibase#9334] [Help]
Hello!

I am trying to use a self-hosted, locally running Budibase with Keycloak 20.0.1 running inside a Docker container.

When I go to http://localhost:10000 I see the following screen:

![image](https://user-images.githubusercontent.com/121552160/212246218-f32c633b-6caa-499c-84ed-9678dcd53fed.png)

When I press the `Sign In with OIDC` button I get the error

```
{"message":"Error constructing OIDC authentication configuration - 
FetchError: request to http://localhost:8080/realms/bb/.well-known/openid-configuration failed, 
reason: connect ECONNREFUSED 127.0.0.1:8080","status":500}
```

However if I open that URL (`http://localhost:8080/realms/bb/.well-known/openid-configuration`) in the browser it does seem to work (see below).

![image](https://user-images.githubusercontent.com/121552160/212247119-59c69790-5541-4ce7-a5d6-4f0c7ded035a.png)

Why does this error occur and how can I fix it?

Thanks in advance

**Comment by melohagan** [comment]
Hey @dpfr2022 

You may need to use `host.docker.internal` instead of `localhost` in your configuration:
https://docs.budibase.com/docs/openid-connect#setup
